tests/test_pytypes.py

```diff
@@ -1048,3 +1048,46 @@ def test_typevar(doc):
     assert doc(m.annotate_listT_to_T) == "annotate_listT_to_T(arg0: list[T]) -> T"
 
     assert doc(m.annotate_object_to_T) == "annotate_object_to_T(arg0: object) -> T"
+
+
+@pytest.mark.skipif(
+    not m.defined_PYBIND11_HAS_RANGES, reason="C++20 <ranges> not available."
+)
+def test_ranges(capture):
+    assert m.iterator_default_initialization
```

Comment:
I think what you want here is `assert m.iterator_default_initialization()`

If you don't call the function you're just asserting that the Python object exists and is truthy.

For all tests you're adding, it's best (and usually easy) to organize them so that they all run independently from each other, as much as possible. That's so that we'll see all failures/successes at once, which can be very helpful. E.g. if `Tuple` fails but `List` and `Dict` pass, vs all of `Tuple`, `List`, `Dict` fail; that is a great clue to have when starting to debug. Usually using `@pytest.mark.parametrize()` is great for that. Look in test_unnamed_namespace_a.py for an example.